ironmon_tracker/screens/CoverageCalcScreen.lua

```diff
@@ -0,0 +1,593 @@
+		for _, pokemonID in ipairs(SCREEN.CoverageData[tab] or {}) do
+			local bst = tonumber(PokemonData.Pokemon[pokemonID].bst or "") or 0
+			local button = {
+				type = Constants.ButtonTypes.POKEMON_ICON,
+				getIconId = function(self) return self.id, SpriteData.Types.Idle end,
+				id = pokemonID,
+				bst = bst,
+				tab = tab,
+				dimensions = { width = 32, height = 32, },
+				boxColors = { SCREEN.Colors.border, SCREEN.Colors.boxFill },
+				isVisible = function(self) return SCREEN.currentView == SCREEN.Views.Pokemon and SCREEN.Pager.currentPage == self.pageVisible end,
+				includeInGrid = function(self) return SCREEN.currentTab == self.tab end,
+				-- Intentionally don't allow clicking on icons yet, as clicking Back isn't smart enough to return to this screen; might as in later
+				-- onClick = function(self)
+				-- 	InfoScreen.changeScreenView(InfoScreen.Screens.POKEMON_INFO, self.id) -- implied redraw
+				-- end,
+				-- Only draw after all the pokemon icons are drawn; executed manually in drawScreen
+				-- drawAfter = function(self, shadowcolor)
+				-- 	local x, y = self.box[1], self.box[2]
+				-- 	local textColor = Theme.COLORS[self.textColor]
+				-- 	local bgColor = Theme.COLORS[self.boxColors[2]]
+				-- 	-- Draw the text below the icon
+				-- 	Drawing.drawTransparentTextbox(x + centeredOffsetX - 1, y + 33, text, textColor, bgColor, shadowcolor)
+				-- end,
+			}
+			table.insert(SCREEN.Pager.Buttons, button)
+		end
+	end
+end
+
+function CoverageCalcScreen.changeTab(tab)
+	SCREEN.currentTab = tab
+	SCREEN.Pager:realignButtonsToGrid()
+	SCREEN.refreshButtons()
+	Program.redraw(true)
+end
+
+function CoverageCalcScreen.prepopulateMoveTypes()
+	SCREEN.resetTypesAndData()
+
+	local leadPokemon = Tracker.getPokemon(1, true)
+	if not leadPokemon then
+		return
+	end
+
+	local allowedCategories = {
+		[MoveData.Categories.PHYSICAL] = true,
+		[MoveData.Categories.SPECIAL] = true,
+	}
+	-- These shouldn't be counted automatically as "coverage" since they don't have a damage multiplier
+	-- Unfortunately this excludes Night Shade when caring about Shedinja, but don't have a good solution otherwise
+	local excludedMoveIds = {
+		[12] = true, -- Guillotine
+		[32] = true, -- Horn Drill
+		[49] = true, -- SonicBoom
+		[68] = true, -- Counter
+		[69] = true, -- Seismic Toss
+		[82] = true, -- Dragon Rage
+		[90] = true, -- Fissure
+		[101] = true, -- Night Shade
+		[117] = true, -- Bide
+		[149] = true, -- Psywave
+		[162] = true, -- Super Fang
+		[165] = true, -- Struggle (I guess)
+		[237] = true, -- Hidden Power
+		[243] = true, -- Mirror Coat
+		[248] = true, -- Future Sight
+		[251] = true, -- Beat Up
+		[283] = true, -- Endeavor
+		[329] = true, -- Sheer Cold
+		[353] = true, -- Doom Desire
+	}
+
+	for _, move in ipairs(leadPokemon.moves or {}) do
+		if MoveData.isValid(move.id) then
+			-- TODO: Might use current moveset to help calc stricter coverage later (e.g. Night Shade, Will-O-Wisp, Hidden Power)
```

Comment:
Yeah, i had considered the current moveset being included in the calculations, but I feel that goes beyond this simple calculator, and might be confusing in certain contexts, since you cant see your actual moveset while on this page. So instead, i went with just the super simple "add relevant types" solution, then letting the player refine their coverage as they see fit.